src/components/expression/expression-loader.js

```diff
@@ -133,40 +148,103 @@ genotet.ExpressionLoader.prototype.loadExpressionMatrix_ = function(fileName,
 
 /**
- * Loads the TFA for selected genes.
+ * Loads the profile for selected genes.
  * @param {string} fileName Name of the expression matrix.
- * @param {!Array<string>} geneNames Names for gene selection.
+ * @param {!Array<string>} geneNames Name for gene selection.
  * @param {!Array<string>} conditionNames Names for experiment condition
  *      selection.
- * @private
+ * @param {boolean} isAddProfile Whether it is adding a new profile.
  */
-genotet.ExpressionLoader.prototype.loadTfaProfile_ = function(fileName,
-    geneNames, conditionNames) {
-  var tfaParams = {
-    type: genotet.expression.QueryType.TFA_PROFILE,
-    fileName: genotet.data.tfaFileName,
-    geneNames: geneNames,
-    conditionNames: conditionNames
+genotet.ExpressionLoader.prototype.loadProfile =
+  function(fileName, geneNames, conditionNames, isAddProfile) {
+    var params = {
+      type: genotet.expression.QueryType.PROFILE,
+      fileName: fileName,
+      geneNames: geneNames,
+      conditionNames: conditionNames
+    };
+    this.get(genotet.data.serverURL, params, function(data) {
+      // Store the last applied data selectors.
+      if (data.geneNames.length == 0 || data.conditionNames.length == 0) {
+        return;
+      }
+      if (isAddProfile) {
+        var geneName = geneNames[0];
+        var geneIndex = this.data.profiles.length;
+        this.data.profileGeneNameDict[geneName] = geneIndex;
+        this.data.profile.geneNames.push(geneName);
+        this.data.profile.values = this.data.profile.values.concat(
+          data.values);
+        this.data.profile.valueMin = Math.min(this.data.profile.valueMin,
+          data.valueMin);
+        this.data.profile.valueMax = Math.max(this.data.profile.valueMax,
+          data.valueMax);
+        this.signal('newProfileLoaded', {
+          geneName: geneName,
+          geneIndex: geneIndex
+        });
+      } else {
+        this.data.profile = data;
+        this.signal('profileLoaded');
+      }
+    }.bind(this), 'cannot load expression profiles', true);
+  };
+
+/**
+ * Loads the TFA for selected genes.
+ * @param {string} fileName Name of the TFA file.
+ * @param {!Array<string>} geneNames Names for gene selection.
+ * @param {!Array<string>} conditionNames Names for TFA experiment condition
+ *      selection.
+ * @param {boolean} isAddProfile Whether it is adding a new TFA profile.
+ */
+genotet.ExpressionLoader.prototype.loadTfaProfile =
+  function(fileName, geneNames, conditionNames, isAddProfile) {
+    var tfaParams = {
+      type: genotet.expression.QueryType.TFA_PROFILE,
+      fileName: genotet.data.tfaFileName,
+      geneNames: geneNames,
+      conditionNames: conditionNames
+    };
+    this.get(genotet.data.serverURL, tfaParams, function(data) {
+      // Store the last applied data selectors.
+      if (data.tfaValues.length == 0) {
+        genotet.warning('TFA not found');
+        return;
+      }
+      if (isAddProfile) {
+        var geneName = geneNames[0];
```

Comment:
Well "you will not add multiple genes" does not mean that your function need to have ambiguous and special handling. If your argument is "geneNames" then expectedly you should be able to add multiple genes using this function (regardless of whether there wil be such use case). Any special handling or ambiguity has to be addressed by comments, or refactored by better method separation.